src/mscorlib/shared/System/Runtime/InteropServices/MemoryMarshal.cs

```diff
@@ -0,0 +1,27 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>
+    /// Provides a collection of methods for interoperating with <see cref="Memory{T}"/>, <see cref="ReadOnlyMemory{T}"/>,
+    /// <see cref="Span{T}"/>, and <see cref="ReadOnlySpan{T}"/>.
+    /// </summary>
+    public static class MemoryMarshal
```

Comment:
It looks like MemoryMarshal.cs and most of Memory.cs/ReadOnlyMemory.cs share their implementation between corelib and System.Memory. Is there a better way to keep the sources in sync or is a comment stating "apply changes to both places" good enough?

Reply:
There are only a few places where there are meaningful differences.  It might be helpful to explore factoring those pieces out into helper methods in another partial class file, such that the main file with most of the logic can be sync'd and kept identical between them, with only the differences in files unique to the repos.

In the case of MemoryMarshal, the AsMemory method I've added here is identical, but I expect there will be other methods on MemoryMarshal that are not, and that will need internals access to the Memory and Span implementations.

Anyway, as a first step, once our active PRs go through, I'd suggest someone compares the files in the two repos and gets them to be as close as possible.  I noticed there are various discrepancies between them that make diff'ing harder, like differences in XML comments, differences in the ordering of members, etc.

Reply:
>  I'd suggest someone compares the files in the two repos and gets them to be as close as possible.

I wanted to do exactly that. If you would like to do so, let me know. Otherwise, I can essentially fix the ordering/comments so they match.

> I noticed there are various discrepancies between them that make diff'ing harder, like differences in XML comments, differences in the ordering of members, etc.

Yep, noticed that myself.

Reply:
We have option of mirroring directories between repos. We have the mirror for shared CoreLib directory (between CoreCLR and CoreRT), and for System.Diagnostics.Tracing directory (between CoreCLR and CoreFX - I have noticed @brianrob set it up recently - you can talk to him about what it involved). We can use the same tech for more parts, like System.Memory. It would get rid of the forwarders from SpanExtension to Span as a side-effect that I am not particularly happy about.

>  It might be helpful to explore factoring those pieces out into helper methods in another partial class file

Ifdefs work great for small differences as well.

Reply:
If you're interested in mirroring, I worked with @safern to get this done.  We have some improvements to make, because the code gets mirrored to multiple repos, but if you don't need your code mirrored to multiple repos then it should be fairly straight forward.

Reply:
https://github.com/dotnet/corefx/issues/25188